src/Http/Http.Abstractions/src/Api/IFromRouteMetadata.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Http.Api
```

Comment:
We don't need any new namespaces. Put it in Http

Reply:
I was considering that. I figured if you're not implementing custom attributes or a filter or something like that. I don't want to noise up intellisense any more than it is already for people who are `using Microsoft.AspNetCore.Http;` just for something basic like `HttpContext`. People writing basic middleware don't need any of this.

Reply:
I think we initially had too many namespaces and I don't want to make that worse, I don't see a need to nest these APIs in another namespace. This is similar to how we have [IAuthorizeMetadata](https://github.com/dotnet/aspnetcore/blob/44106c887c1475f2f2ef2a2489af416bf8a57064/src/Http/Metadata/src/IAuthorizeData.cs#L4) in the root authorization namespace.

Reply:
I think we should default to that, and then we can duke it out in API review, if we need more namespaces.